generated/nidaqmx/system/watchdog.py

```diff
@@ -538,7 +540,7 @@ def close(self):
         within the loop after you are finished with the task to avoid
         allocating unnecessary memory.
         """
-        if self._handle is None:
+        if self._handle is None or not self._is_new_session_initialized:
```

Comment:
`_is_new_session_initialized` being `False` isn't a failure case. The user may have specifically requested to attach to an existing task rather than create a new one. They can close that, if they like. The alternative to close is detach, which "leaks" the task on the server.

Reply:
I think you can just remove this check

Reply:
Confirmed, just remove it. If someone explicitly calls `close`, then we should close.

Reply:
Removed the conditional check for close on exit.